scale_build/image/iso.py

```diff
@@ -34,8 +34,15 @@ def install_iso_packages_impl():
     for package in get_manifest()['iso-packages']:
         run_in_chroot(['apt', 'install', '-y', package])
 
+    # Inject vendor name into grub.cfg
+    with open(CONF_GRUB, 'r') as f:
+        grub_cfg = f.read()
+    if TRUENAS_VENDOR:
+        grub_cfg = grub_cfg.replace('TrueNAS SCALE', TRUENAS_VENDOR)
```

Comment:
Too broad, we can end up with unexpected substitutions. Can we put `TRUENAS_VENDOR` during the config file generation?

Reply:
The config under `boot/grub/grub.cfg` is copied over in this function anyway, I'm just modifying it if relevant.

Reply:
Can we modify default `grub.cfg` to have something like `$vendor` here?

Reply:
I'm not sure how I'd go about setting up that variable in the first place, but if we can then that would be the better solution. Later in the file there's some grub.cfg injection similar to what I'm doing (with the string replace and all), so I modeled my changes after that

Reply:
Look what `CONF_GRUB` variable is, it points to a file in this repository

Reply:
I mean how I'd set it within the .cfg itself. I think I'd need to check for the .vendor file, parse the JSON, then use that. If that's the intention then I will do that

Reply:
You just edit cfg file in the repository and replace `Start TrueNAS SCALE Installation` with `Start $vendor Installation` so that we can reliably replace `$vendor` later

Reply:
With those changes, where would we set that variable so it's picked up in the `grub.cfg`? Not super familiar with the grub config process, especially with things like those external variables.

Reply:
It will not be picked up, you'll have to replace it the same way yo do now

Reply:
Ohhhhhh okay I see what you're suggesting now.

So instead of looking for `TrueNAS SCALE` and doing the substring replacement with that, we instead have `$vendor` as a placeholder and we look for that and replace accordingly. Defaulting to TrueNAS SCALE.